### PR TITLE
TECH_DEBT: Adding documentation for security

### PR DESCRIPTION
**Pull request by seanmcilvenna**
### 🛠️ Description of Changes

- **Documentation**
	- Updated the README to include a new "Kafka Retry Topics" entry and a new "Security" section with links to detailed documents.
	- Introduced new documents: 
		- `oauth_cookie_flow.md` outlining the OAuth 2.0 authentication flow.
		- `security_overview.md` providing an overview of the security model and Role-Based Access Control (RBAC).

### 🧪 Testing Performed

No testing necessary. Documentation updates only.

### 📓 Documentation Updated

Pages added for security.

## Summary by CodeRabbit

- **New Features**
	- Introduced a "Security" section in the documentation, enhancing organization and clarity.
	- Added comprehensive documentation on security overview and OAuth authentication flow.

- **Documentation**
	- Updated README to include links to new security-related documents.
	- Created detailed documentation files for "Security Overview" and "OAuth & Cookie Flow". 

- **Bug Fixes**
	- Removed duplicate entry for "Kafka Retry Topics" from the README.

### CODE REVIEW

File: docs/images/security.jpg
Comment (sdmcgeown) [inline docs/images/security.jpg:1]: In the diagram, it says the cookie lifetime is configurable with a default of 30 minutes, but I think it's supposed to be a session cookie

Comment (amphillipsLGC) [reply docs/images/security.jpg:1]: I need to update that. Good catch

Comment (amphillipsLGC) [reply docs/images/security.jpg:1]: Updated diagram
LinkAdminSecurity

**Review by amphillipsLGC — APPROVED** [review]
looks good

File: docs/functionality/oauth_flow.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+      "Endpoints": {
+        "Authorization": "",
+        "Token": "",
+        "UserInformation": ""
+      },
+      "CallbackPath": "/api/signin-oauth2"
+    }
+  }
+}
+```
```
[the remark this answers is not in the document]
Comment (seanmcilvenna) [reply docs/functionality/oauth_flow.md:134]: Ignoring for now.

**Pull request merged**